src/pages/User/VerifyKey.jsx

```
@@ -0,0 +1,79 @@
+import isEmpty from "lodash/isEmpty";
+import { Message } from "pi-ui";
+import qs from "query-string";
+import React, { useEffect } from "react";
+import { withRouter } from "react-router-dom";
+import SingleContentPage from "src/componentsv2/layout/SingleContentPage";
+import { verifyUserPubkey } from "src/helpers";
+import useVerifyKey from "src/hooks/api/useVerifyKey";
+import * as pki from "src/lib/pki";
+
+const VerifyKey = ({ location, history }) => {
+  const {
+    apiMeResponse,
+    loggedInAsEmail,
+    userPubkey,
+    keyMismatchAction,
+    verifyUserKey,
+    updateMe,
+    verifyUserKeyError,
+    onVerifyUserKey
+  } = useVerifyKey();
+
+  useEffect(() => {
+    const { verificationtoken } = qs.parse(location.search);
+    if (
+      isEmpty(location.search) ||
+      !verificationtoken ||
+      typeof verificationtoken !== "string"
+    ) {
+      history.push("/user/login");
```

Better throw an error here instead of redirecting to login without further explanation.

---

Yeah, I agree.